src/backends/perf_pt/collect.c

```diff
@@ -142,6 +163,84 @@ bool perf_pt_start_tracer(struct tracer_ctx *, struct perf_pt_trace *, struct pe
 bool perf_pt_stop_tracer(struct tracer_ctx *tr_ctx, struct perf_pt_cerror *);
 bool perf_pt_free_tracer(struct tracer_ctx *tr_ctx, struct perf_pt_cerror *);
 
+
+/*
+ * Reads samples from the data buffer and acts upon them appropriately.
+ *
+ * Returns true on success, or false otherwise.
+ */
+static bool
+handle_sample(void *aux_buf, struct perf_event_mmap_page *hdr,
+         struct perf_pt_trace *trace, void *data_tmp,
+         struct perf_pt_cerror *err)
+{
+    bool ret = true;
```

Comment:
Given that the `clean:` target just does `return ret`, I think this variable, and the goto label, are obfuscatory.

Reply:
I think you are right. An earlier version probably had cleanup, which is why this has happened.